plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/util/SolidFireUtil.java

```diff
@@ -397,65 +508,290 @@ public static void hostAddedToOrRemovedFromCluster(long hostId, long clusterId,
         }
     }
 
-    public static long placeVolumeInVolumeAccessGroup(SolidFireConnection sfConnection, long sfVolumeId, long storagePoolId,
-                                                      String vagUuid, List<HostVO> hosts, ClusterDetailsDao clusterDetailsDao) {
-        if (hosts == null || hosts.isEmpty()) {
-            throw new CloudRuntimeException("There must be at least one host in the cluster.");
+    // Put the host in an existing VAG or create a new one (only create a new one if all existing VAGs are full (i.e. 64 hosts max per VAG) and if
+    // creating a new VAG won't exceed 4 VAGs for the computer cluster).
+    // If none of the hosts in the cluster are in a VAG, then leave this host out of a VAG.
+    // Place applicable volume IDs in VAG, if need be (account of volume starts with SF_CS_ACCOUNT_PREFIX).
+    private static void handleVagForHost(SolidFireUtil.SolidFireConnection sfConnection, List<SolidFireUtil.SolidFireVag> sfVags, Host host, HostDao hostDao) {
```

Comment:
Can you use proper java doc notation? Those notations there `//` are for code comments.

Reply:
So, that's an interesting question. In this case, these comments are aimed at a developer who is looking for help in understanding the logic of the method as opposed to a developer who is looking to simply invoke the method to have some work done for him/her. I'm OK with using JavaDoc, if you feel that makes more sense here, but I figured I would clarify to you first what the intent of these comments is. Please let me know if you'd still like to see this in JavaDoc form. Thanks!

Reply:
You have a clear definition of the difference between Javadocs and java comments. That is awesome 👍 

I asked because I have seen people trying to write documentation, but they used comment notation instead. If you believe it is ok with Java comment notation, I am ok with it ;)